src/components/molecules/ImageInput/ImageInput.tsx

```diff
@@ -24,94 +24,82 @@ interface ImageInputProps extends ContainerClassName {
-      handleFileInputChange,
-    } = useImageInputCompression(register, error?.message, name);
+  const {
+    loading,
+    errorMessage,
+    handleFileInputChange,
+  } = useImageInputCompression(register, error?.message, name);
 
-    const handleFileInputChangeWrapper = useCallback(
-      async (event: ChangeEvent<HTMLInputElement>) => {
-        const [url, compressedFile] = await handleFileInputChange(event);
-        if (!compressedFile || !url) return;
+  const handleFileInputChangeWrapper = useCallback(
+    async (event: ChangeEvent<HTMLInputElement>) => {
+      const [url, compressedFile] = await handleFileInputChange(event);
+      if (!compressedFile || !url) return;
 
-        setValue(name, [compressedFile], false);
+      setValue(name, [compressedFile], false);
 
-        if (remoteUrlInputName) {
-          setValue(remoteUrlInputName, url, true);
-        }
-      },
-      [handleFileInputChange, name, remoteUrlInputName, setValue]
-    );
+      if (remoteUrlInputName) {
+        setValue(remoteUrlInputName, url, true);
+      }
+    },
+    [handleFileInputChange, name, remoteUrlInputName, setValue]
+  );
 
-    return (
-      <>
-        <div
-          className={classNames(
-            `image-input default-container`,
-            containerClassName,
-            {
-              disabled: loading,
-            }
-          )}
-        >
-          {imageUrl ? (
-            <img
-              className={`default-image ${imageClassName}`}
-              src={imageUrl}
-              alt="upload"
-            />
-          ) : (
-            <div className="centered-flex empty">
-              <h6 className=" text" style={{ fontWeight: "lighter" }}>
-                Click to upload an image
-              </h6>
-            </div>
-          )}
-          <input
-            disabled={disabled || loading}
-            type="file"
-            accept={ACCEPTED_IMAGE_TYPES}
-            onChange={handleFileInputChangeWrapper}
-            className="default-input"
+  return (
+    <>
+      <div
+        className={classNames(
+          `image-input default-container`,
+          containerClassName,
+          {
+            disabled: loading,
+          }
+        )}
+      >
+        {imageUrl ? (
+          <img
+            className={`default-image ${imageClassName}`}
+            src={imageUrl}
+            alt="upload"
           />
-          {loading && <ImageOverlay>processing...</ImageOverlay>}
-          {remoteUrlInputName && (
-            <input
-              type="hidden"
-              ref={ref}
-              name={remoteUrlInputName}
-              value={remoteImageUrl}
-            />
-          )}
-        </div>
-        {errorMessage && <span className="input-error">{errorMessage}</span>}
-      </>
-    );
-  }
-);
-
-ImageInput.displayName = "ImageInput";
+        ) : (
+          <div className="centered-flex empty">
+            <h6 className=" text" style={{ fontWeight: "lighter" }}>
```

Comment:
Could probably be useful to replace things like `<h6>` that are with semantical meaning but used only for "free style reuse" with `<div>` or `<span>` having the appropriate styles set.

Reply:
https://github.com/sparkletown/sparkle/pull/2201#discussion_r700168039